R/workflow.R

```diff
@@ -276,13 +277,26 @@ copy_nonmem_model_from <- function(
 
   # copy control steam to new path
   if (.update_mod_file) {
+    parent_mod_id <- .parent_spec %>% get_model_path() %>% get_mod_id()
+    new_mod_id <- .new_model %>% get_mod_id()
+
     # read parent control stream
     mod_str <- .parent_spec %>% get_model_path() %>% read_file()
 
     # replace the $PROBLEM line(s)
     mod_str <- str_replace(mod_str,
                 "\\$PROB(.|\n)*?\\$",
-                as.character(glue("$PROBLEM {get_mod_id(.new_model)} {.description}\n\n$")))
+                as.character(glue("$PROBLEM {new_mod_id} {.description}\n\n$")))
+
+    # boilerplate code from Sam to replace other occurances of model id #### !!!!!!
+    # need to test and refactor to use stringr
```

Comment:
look at str_replace_all(). That said, I don't think this is the right place to have this. It is too metrum specific and will lock this function into a specific api. These replacement functions should be separate and modular pieces `replace_mod_references(spec, ...)` or the like. For example, what happens in bayesian land when you also have some other types of files, or use PSN conventions and want sdtab, patab, etc. 

These are good helpers to have, but just not here